backend-lib/src/main/rust/lib.rs

```diff
@@ -1512,16 +1542,28 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_proposeSh
                             e
                         )
                     })
-            })?
-            .into_keys()
-            .collect();
+            })?;
+
+        let from_addrs = if let Some((addr, _)) = transparent_receiver.map_or_else(||
+            if account_receivers.len() > 1 {
+                Err(format_err!(
+                    "Account has more than one transparent receiver with funds to shield; this is not yet supported by the SDK. Provide a specific transparent receiver to shield funds from."
+                ))
+            } else {
+                Ok(account_receivers.iter().next().map(|(a, v)| (*a, *v)))
+            },
+            |addr| Ok(account_receivers.get(&addr).map(|value| (addr, *value)))
+        )?.filter(|(_, value)| *value >= shielding_threshold.into()) {
+            [addr]
+        } else {
+            // There are no transparent funds to shield; don't create a proposal.
+            return Ok(ptr::null_mut());
```

Comment:
I don't agree that we should use `Ok(ptr::null_mut())` here, as opposed to implementing this via an error return. It essentially means that any `FfiBoxedSlice` could wrap a null pointer, which imposes a reliability cost on all of its uses, when it isn't necessary for most of them.

Reply:
You're misunderstanding this return type. The Android SDK does not have an `FfiBoxedSlice`; that is a Swift-specific type as that uses C FFI. Here we have the JNI, and construct a `ByteArray?` directly; the `ptr::null_mut()` here directly maps to a Java `null`, hich is the correct value for the `None` case of a nullable type like `ByteArray?`.